controllers/controlplane_controller.go

```diff
@@ -139,6 +138,49 @@ func (r *ControlPlaneReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		r.Log.Info(fmt.Sprintf("PasswordSecret %s exists", instance.Spec.PasswordSecret))
 	}
 
+	ipsetDetails := common.IPSet{
+		Networks:            instance.Spec.Controller.Networks,
+		Role:                controlplane.Role,
+		HostCount:           controlplane.Count,
+		VIP:                 true,
+		AddToPredictableIPs: false,
+	}
+	ipset, op, err := common.OvercloudipsetCreateOrUpdate(r, instance, ipsetDetails)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("IPSet for %s successfully reconciled - operation: %s", instance.Name, string(op)))
+	}
+
+	if len(ipset.Status.HostIPs) != controlplane.Count {
```

Comment:
I've seen this IP vs count `!=` check cause problems on scale-down for `BaremetalSet` and `VMSet`.  Switching to `<` seemed to fix the isssue.  Might need that here as well, unless the IPs are in fact being removed from the `OvercloudIPSet` during scale-down.

Reply:
changed